src/model/page-config-resolver-test.js

```diff
@@ -268,7 +272,350 @@ describes.realWin('PageConfigResolver', {}, env => {
   });
 
   describe('parse microdata', () => {
-    // TODO(sohanirao): Add tests to parse microdata
+    let readyState;
+
+    beforeEach(() => {
+      readyState = 'loading';
+      Object.defineProperty(doc, 'readyState', {get: () => readyState});
+    });
+
+    it('should parse locked access', () => {
+      const divElement = createElement(doc, 'div');
+      divElement.innerHTML =
+          '<div itemscope itemtype="http://schema.org/NewsArticle"> \
+            <meta itemprop="isAccessibleForFree" content="False"/> \
+            <div itemprop="isPartOf" itemscope itemtype="http://schema.org/CreativeWork http://schema.org/Product"> \
+              <meta itemprop="name" content="New York Times"/> \
+              <meta itemprop="productID" content="pub1:premium"/> \
+            </div> \
+            <div itemprop="articleBody" class="paywalled-section"> \
+              Paid content possibly. \
+            </div> \
+          </div>';
+      addMicrodata(divElement);
+      readyState = 'complete';
+      const resolver = new PageConfigResolver(gd);
+      return resolver.resolveConfig().then(config => {
+        expect(config.isLocked()).to.be.true;
+        expect(config.getProductId()).to.equal('pub1:premium');
+      });
+    });
+
+    it('should parse unlocked access', () => {
+      const divElement = createElement(doc, 'div');
+      divElement.innerHTML =
+          `<div itemscope itemtype="http://schema.org/NewsArticle">
+            <div itemprop="isPartOf" itemscope itemtype="http://schema.org/CreativeWork http://schema.org/Product">
+              <meta itemprop="name" content="New York Times"/>
+              <meta itemprop="productID" content="pub1:premium"/>
+            </div>
+            <div>
+              <meta itemprop="isAccessibleForFree" content="true"/>
+              <meta itemprop="cssSelector" content=".paywalled-section"/>
+            </div>
+            <div itemprop="articleBody" class="paywalled-section">
+              Paid content possibly.
+            </div>
+          </div>`;
+      addMicrodata(divElement);
+      readyState = 'complete';
```

Comment:
I'd generally reduce `readyState = 'complete'` use to only few tests where it really matters.

Reply:
Done